src/referentielijsten_client/steps.py

```diff
@@ -0,0 +1,61 @@
+from django.core.exceptions import ValidationError
+
+import structlog
+from django_setup_configuration.configuration import BaseConfigurationStep
+from django_setup_configuration.exceptions import ConfigurationRunFailed
+from zgw_consumers.models import Service
+
+from openklant.config.models import ReferentielijstenConfig
+from referentielijsten_client.models import ReferentielijstenConfigurationModel
+
+logger = structlog.stdlib.get_logger(__name__)
+
+
+class ReferentielijstenConfigurationStep(
+    BaseConfigurationStep[ReferentielijstenConfigurationModel]
+):
+    namespace = "referentielijsten"
+    enable_setting = "REFERENTIELIJSTEN_CONFIG_ENABLE"
+
+    verbose_name = "Configuration for Referentielijsten service"
+    config_model = ReferentielijstenConfigurationModel
+
+    def execute(self, model: ReferentielijstenConfigurationModel) -> None:
+        logger.info(
+            "configuring_referentielijsten",
+            enabled=model.enabled,
+            api_root=model.service_api_root,
+        )
+
+        try:
+            service, created = Service.objects.update_or_create(
+                api_root=model.service_api_root,
+                defaults={
+                    "label": "Referentielijsten",
+                },
+            )
+            if created:
+                logger.info("service_created", api_root=model.service_api_root)
+            else:
+                logger.debug("service_updated", api_root=model.service_api_root)
+
+            config_instance = ReferentielijstenConfig.get_solo()
+            config_instance.enabled = model.enabled
+            config_instance.service = service
+            config_instance.tabel_code = model.tabel_code
+
+            config_instance.full_clean()
+            config_instance.save()
+
+            logger.info("referentielijsten_config_updated", success=True)
+
+        except ValidationError as exc:
+            logger.error("configuration_validation_failed", exc_info=exc)
+            raise ConfigurationRunFailed(
+                f"Validation failed while configuring Referentielijsten: {exc}"
+            ) from exc
+        except Exception as exc:
+            logger.error("configuration_failed", exc_info=exc)
+            raise ConfigurationRunFailed(
+                f"Failed to configure Referentielijsten: {exc}"
+            ) from exc
```

Comment:
I think the try-except block can be removed if the `full_clean` is removed as well
```suggestion
```